### PR TITLE
Fixes #14

### LINKED ISSUE #14: could not build server_names_hash, you should increase server_names_hash_bucket_size
I have an application which uses a long domain name. At ssl-proxy startup, I get the following error message

```
https_offloader_1  | 2019/11/07 10:28:53 [emerg] 13#13: could not build server_names_hash, you should increase server_names_hash_bucket_size: 64
https_offloader_1  | nginx: [emerg] could not build server_names_hash, you should increase server_names_hash_bucket_size: 64
```

And ssl-proxy shuts down.

According to [stackoverflow](https://stackoverflow.com/a/13906493/15619), this may be due to a "bad" configuration.

### PR DESCRIPTION
**Pull request by Riduidel**
Added customizable server_names_hash_size (useful for long domain names)

### CODE REVIEW

File: nginx/entrypoint.sh [hunk 1]
```diff
@@ -92,6 +93,7 @@ events {
 pid         /var/run/nginx.pid;
 
 http {
+  server_names_hash_bucket_size  $SERVER_NAMES_HASH_SIZE;
```
Comment (justsml) [inline nginx/entrypoint.sh:98]: This looks like it's missing an escaped `$`

I think it needs to be like: `server_names_hash_bucket_size  \$SERVER_NAMES_HASH_SIZE;`

File: nginx/entrypoint.sh [hunk 2]
```diff
@@ -15,6 +15,7 @@ CERT_PUBLIC_PATH=${CERT_PUBLIC_PATH-"/certs/fullchain.pem"}
 CERT_PRIVATE_PATH=${CERT_PRIVATE_PATH-"/certs/privkey.pem"}
 HTTPS_PORT=${HTTPS_PORT-"443"}
 TLS_PROTOCOLS=${TLS_PROTOCOLS-"TLSv1 TLSv1.1 TLSv1.2"}
+SERVER_NAMES_HASH_SIZE=${SERVER_NAMES_HASH_SIZE-"32"}
```
Comment (justsml) [inline nginx/entrypoint.sh:18]: I think there should be a longer default, at least 64. Probably 128.

Did you find any reasons against it?

File: README.md [hunk 3]
```diff
@@ -168,6 +168,7 @@ services:
 |PASSWORD           |               | Both PASSWORD and USERNAME must be set in order to use Basic authorization
 |PASSWD_PATH        | /etc/nginx/.htpasswd | Alternate auth support (don't combine with USERNAME/PASSWORD) Bind-mount a custom path to `/etc/nginx/.htpasswd`
 |ADD_HEADER         | Not set       | Useful for tagging routes in your infrastructure.
+|SERVER_NAMES_HASH_SIZE         | 32       | Variable conditionning maximum size of server name. Set it to 64/128/... if nginx fails to start with `could not build server_names_hash, you should increase server_names_hash_bucket_size` error message.
```
Comment (justsml) [inline README.md]: I'd split this into 2 parts....


First for the `SERVER_NAMES_HASH_SIZE`'s Notes field, try this: "Configures size of server name hash. Suggested values include 64, 128, 256, etc."

=================================

_This next bit should probably be under a troubleshooting section_

## Troubleshooting
* Failing to start
    * "could not build server_names_hash"
    If nginx fails to start with `could not build server_names_hash, you should increase server_names_hash_bucket_size` error message, you might need to configure environment variable: [`SERVER_NAMES_HASH_SIZE`](#arguments).

**Comment by justsml** [comment]
## Thanks for the PR @Riduidel !!! 😄 

Please let me know if you have any questions about my notes.

If they sound like reasonable suggestions I will get this merged ASAP. 🤘

File: nginx/entrypoint.sh (hunk 1 above)
Comment (Riduidel) [reply nginx/entrypoint.sh:98]: Well, it seems to work on my branch without the `\` ... so I would leave it this way, no ?

File: nginx/entrypoint.sh (hunk 2 above)
Comment (Riduidel) [reply nginx/entrypoint.sh:18]: The value of `32` is the default value used by nginx, so i try to comply with existing code as much as possible.

File: README.md (hunk 3 above)
Comment (Riduidel) [reply README.md]: Good idea!

**Comment by Riduidel** [comment]
Well, if you can take care of the points you mentionned, I have no time currently to do it myself currently. Thanks for the review!

**Comment by justsml** [comment]
Thanks again, Nicolas!
I'll spend some time over the upcoming holidays doing a ton of Open Source
housekeeping. 💪

ssl-proxy is high on my list.

**Pull request merged**